### ISSUE TITLE
[BUG] Tag change removes tc-storyview-zoomin-tiddler class from tiddler frame

### ISSUE THREAD

**Issue by yaisog** [Jermolene/TiddlyWiki5#7247]
### Describe the bug

When tags are added via `$fieldmangler`, the `tc-storyview-zoomin-tiddler` class which is present for all tiddlers in the zoomin storyview is removed from the current tiddler.
Switching back and forth between tiddlers adds the class again.
Removing the class this way can lead to unwanted changes in the UI if there is CSS defined on `tc-storyview-zoomin-tiddler`.

### Expected behavior

Only the class relating to the removed tag should be removed, not the class `tc-storyview-zoomin-tiddler`.

### To Reproduce

Go to tiddlywiki.com and create a tiddler with this content:
```
<$fieldmangler>
<$button message="tm-add-tag" param="HelloThere" class="tc-remove-tag-button">Add</$button>
<$button message="tm-remove-tag" param="HelloThere" class="tc-remove-tag-button">Remove</$button>
</$fieldmangler>
```
Click one of the buttons and observe the classes on the div.tc-tiddler-frame.

### TiddlyWiki Configuration

TW 5.2.5

**Comment by yaisog** [comment]
The same also happens when using the $action-listops widget. Try this in a tiddler in zoomin view:
```
!!Listops
<$button class="tc-remove-tag-button" actions="""<$action-listops $tags="HelloThere" />""">Add</$button>
<$button class="tc-remove-tag-button" actions="""<$action-listops $tags="-[[HelloThere]]" />""">Remove</$button>
```
Changes to other fields than tags do not seem to cause a problem.
I have changed the title of the issue to reflect this.

Looks like the class update of the tiddler-frame div to reflect the change in tags messes with the zoomin class.

**Comment by yaisog** [comment]
As a short term fix to avoid unwanted visual changes, I added this filter run to the class filter in my (custom) ViewTemplate:
```
[{[$:/view}match[zoomin]then[tc-storyview-zoomin-tiddler]]
```
That way, all tiddler-divs always have this class when `zoomin` is active. This should be what zoomin.js does when calling the addClass() method anyway...

**Comment by Jermolene** [comment]
Hi @yaisog as you've discovered, the way that the zoomin story view assigns the `tv-storyview-zoomin-tiddler` class directly to its child DOM nodes is brittle because it ignores the refresh process. That's undoubtedly a bug, and a long-standing one, but it wouldn't be a simple fix. Meanwhile, there may be an alternative approach that you can use: the current storyview is reflected in the classes on the page container element. For example, the default classic storyview assigns the class `tc-page-view-classic` and zoomin assigns `tc-page-view-zoomin`. So perhaps you could target the tiddlers as `.tc-page-container.tc-page-view-zoomin .tc-story-river .tc-tiddler-frame`?

**Comment by yaisog** [comment]
> Meanwhile, there may be an alternative approach that you can use

Hmm, these CSS declarations were not set by me, but in `$:/themes/tiddlywiki/vanilla/base`:
```
	body.tc-body .tc-storyview-zoomin-tiddler {
		width: 100%;
		width: calc(100% - 42px);
	}
	body.tc-body .tc-storyview-zoomin-tiddler {
		width: 100%;
		width: calc(100% - 84px);
	}
.tc-storyview-zoomin-tiddler {
	position: absolute;
	display: block;
	width: 100%;
}
@media (min-width: <<sidebarbreakpoint>>) {
	.tc-storyview-zoomin-tiddler {
		width: calc(100% - 84px);
	}
}
```
The first two also within a @media query and an `if-fluid-fixed` macro. So, to implement your approach would need an update of the core stylesheet.
Meanwhile, I find my approach not half bad. Functionality-wise it does exactly what it is supposed to. I think it could even make all these addClass() calls in zoomin.js superfluous while being compatible to the way all the other classes (e.g. tag-classes) are set on the tiddler frame – all the CSS code in one place.
One could even make it more universal by using something like this:
```
[{[$:/view}addprefix[tc-storyview-]addsuffix[-tiddler]]
```

**Comment by Jermolene** [comment]
Hi @yaisog I see – presumably the absence of the tc-storyview-zoomin-tiddler class causes issues with the width of those tiddlers. Is that what you've observed? I'll push a potential fix momentarily.

**Comment by yaisog** [comment]
Hello @Jermolene, yeah that was what was irritating me. Whenever I changed a tag via a custom button, the width of the tiddler changed by 42px I think.

**Comment by Jermolene** [comment]
Thanks @yaisog – there's a proposed fix at #7252
